pkg/providers/version/version.go

```diff
@@ -51,14 +67,77 @@ func (p *Provider) Get(ctx context.Context) (string, error) {
 	if version, ok := p.cache.Get(kubernetesVersionCacheKey); ok {
 		return version.(string), nil
 	}
-	serverVersion, err := p.kubernetesInterface.Discovery().ServerVersion()
-	if err != nil {
-		return "", err
+	var version string
+	var err error
+	// If we're running locally, these environment variables will be empty.
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+	if len(host) == 0 || len(port) == 0 {
+		// If we're running locally, we don't care which APIServer we hit
+		serverVersion, err := p.kubernetesInterface.Discovery().ServerVersion()
+		if err != nil {
+			return "", err
+		}
+		version = fmt.Sprintf("%s.%s", serverVersion.Major, strings.TrimSuffix(serverVersion.Minor, "+"))
+	} else {
+		// If we're running in cluster, we'll ping each APIServer endpoint to get the minimum k8s version.
+		version, err = p.GetMinKubernetesVersion(ctx)
+		if err != nil {
+			return "", err
+		}
 	}
-	version := fmt.Sprintf("%s.%s", serverVersion.Major, strings.TrimSuffix(serverVersion.Minor, "+"))
 	p.cache.SetDefault(kubernetesVersionCacheKey, version)
 	if p.cm.HasChanged("kubernetes-version", version) {
 		logging.FromContext(ctx).With("version", version).Debugf("discovered kubernetes version")
 	}
 	return version, nil
 }
+
+// GetMinKubernetesVersion ensures that we query all known APIServers for the K8s version.
+// This to handle any scenarios where there may be multiple APIServer reporting different
+// K8s versions.
+func (p *Provider) GetMinKubernetesVersion(ctx context.Context) (string, error) {
```

Comment:
Should `Get()` just handle this for me so I don't have to make different calls to the provider to understand the difference in this behavior?

Reply:
Yeah this is only needed if we're using the http client. I'll make it not public so that users don't have to understand this. Sound good?